core/iwasm/common/wasm_runtime_common.c

```diff
@@ -2364,13 +2374,31 @@ wasm_set_exception_with_id(WASMModuleInstance *module_inst, uint32 id)
         wasm_set_exception(module_inst, "unknown exception");
 }
 
+// TODO(eloparco): Change signature to receive exception buffer
 const char *
```

Comment:
The thread sanitizer complains since this function is used inside the signal handler and it's not safe because it uses `malloc`.

It should be replace by something like
```c
bool
wasm_get_exception(WASMModuleInstance *module_inst, char *exception)
{
    int ret = true;

#if WASM_ENABLE_SHARED_MEMORY != 0
    WASMSharedMemNode *node =
        wasm_module_get_shared_memory((WASMModuleCommon
        *)module_inst->module);
    if (node)
        os_mutex_lock(&node->shared_mem_lock);
#endif
    if (module_inst->cur_exception[0] != '\0') {
        if (exception != NULL)
            bh_memcpy_s(exception, sizeof(module_inst->cur_exception),
                        module_inst->cur_exception,
                        sizeof(module_inst->cur_exception));
    }
    else {
        ret = false;
    }
#if WASM_ENABLE_SHARED_MEMORY != 0
    if (node)
        os_mutex_unlock(&node->shared_mem_lock);
#endif

    return ret;
}
```

but that would require to change multiple APIs, e.g.
```c
WASM_RUNTIME_API_EXTERN bool
wasm_runtime_get_exception(wasm_module_inst_t module_inst, char *exception);

bool
wasm_get_exception(WASMModuleInstance *module, char *exception);

bool
aot_get_exception(AOTModuleInstance *module_inst, char *exception);
```
so I'm waiting to get some feedback before proceeding.

Reply:
Not sure which signal handler uses this function? It is used widely used and had better not change it.
If needed, we can add another API for thread manager, e.g.:
```C
void wasm_copy_exception(WASMModuleInstance *module_inst,
                         char *exception_buf, uint32 exception_buf_size);
```

Reply:
Yes, I came to the same conclusion in the end, better to add a new API.

Reply:
> Not sure which signal handler uses this function?

here
https://github.com/bytecodealliance/wasm-micro-runtime/blob/9f0c4b63ac21c975aee7aae30236e306c8e49d13/core/iwasm/common/wasm_runtime_common.c#L197

Reply:
```c
void wasm_copy_exception(WASMModuleInstance *module_inst,
                         char *exception_buf, uint32 exception_buf_size);
```

I only changed it to return `bool` so that `false` is returned if there is no exception. And `NULL` can be passed as `exception_buf` if the caller is only interested in the returned value (i.e. if an exception has been raised or not). I wanted to avoid creating a separate API like `wasm_has_exception()` to do the same.